### ISSUE TITLE
Go to CSP & SOP with strict compatibility

### ISSUE THREAD

**Issue by narcisgarcia** [LibreTime/libretime#1199]
Web interface uses tracking resources from googleapis.com , gstatic.com and maybe others.
Please let web interface don't depend on third parties. This is for visitors trusting and privacy. Legal concern too.

**Comment by Robbt** [comment]
I support removing these, I believe any of this was a result of legacy code from the pre-fork Airtime.pro branch and so someone would just need to go through and identify spots where the code is linked.

**Comment by paddatrapper** [comment]
I agree that removing these would be a good idea. They aren't used anywhere either

**Comment by zklosko** [comment]
I believe some of our fonts come from Google's font library, hence the calls. The only Google code we should have in is AngularJS for the podcast screens.

**Comment by narcisgarcia** [comment]
What is AngularJS and why is needed there third-party-sourced code?
Strict CSP & SOP guarantees security for users and radio managers.

**Comment by Robbt** [comment]
AngularJS is a JavaScript based web framework. It could be in theory moved to a locally distributed copy. Removing it would require rewriting the podcast code. This is something we are certainly considering as we move forward with a API 2.0

Downloading a local copy of it vs. sourcing it from a Google CDN seems like a good idea to me. We should have LibreTime run w/o requiring any external internet access and without hitting up 3rd party servers for code.

I've not looked into any parts of the code where these lines are included yet though so I don't know how simple it would be.

**Comment by jooola** [comment]
The go to font repository is https://github.com/fontsource/fontsource

This could be installed using yarn/npm and some bundler if any is present.

**Comment by narcisgarcia** [comment]
Making it less 3rd dependent (such as _fontsource_) it should be in installation manual steps.
And moving AngularJS to local copy it should be a prioritized task to get a solid software in this context' terms.

**Comment by jooola** [comment]
> Making it less 3rd dependent (such as fontsource) it should be in installation manual steps.
And moving AngularJS to local copy it should be a prioritized task to get a solid software in this context' terms.

LibreTime does not depend on 3rd party dependencies anymore.

**Comment by narcisgarcia** [comment]
Do you mean 3.0.2 version does not refer to googleapis.com , gstatic.com resources? No external fonts? No external AngularJS?
